codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanDecoder.java

```diff
@@ -230,15 +230,14 @@ AsciiString end() throws Http2Exception {
         }
 
         private void append(int i) {
-            try {
-                bytes[index] = (byte) i;
-            } catch (IndexOutOfBoundsException ignore) {
+            if (bytes.length >= index) {
                 // Always just expand by INITIAL_SIZE
                 byte[] newBytes = new byte[bytes.length + initialCapacity];
                 System.arraycopy(bytes, 0, newBytes, 0, bytes.length);
                 bytes = newBytes;
-                bytes[index] = (byte) i;
             }
+            bytes[index] = (byte) i;
```

Comment:
Consider doing the following:

```java
bytes[index++] = (byte) i;
```

this would be consistent with https://github.com/netty/netty/pull/6845